### PR TITLE
reorganized contract

### PR DESCRIPTION
**Pull request by dbfreem**
### How was it fixed?

Reorganized contract to align with the format discussed in #2441 

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

### CODE REVIEW

File: web3/contract/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from web3.contract.async_contract import (  # noqa: F401
+    AsyncContract,
+    AsyncContractCaller,
+)
+
+from web3.contract.contract import (  # noqa: F401
+    Contract,
+    ContractConstructor,
+    ContractCaller,
+)
```
Comment (kclowes) [inline web3/contract/__init__.py:10]: These are the only classes that get exported to other files in our codebase. I _think_ that our public API all stems from either `Contract` or `AsyncContract`, but I wonder if it's worth exposing other classes like `ContractFunctions` or `ContractEvents` here too?

**Review by kclowes — COMMENTED** [review]
@fselmo / @pacrob  This is ready for a once-over. I'll go through tomorrow and double check that all methods from the old `contract.py` file were copied over, but I'm reasonably confident that they did since the tests are passing.

File: web3/contract/__init__.py (hunk 1 above)
Comment (pacrob) [reply web3/contract/__init__.py:10]: I would lean towards only exposing what is necessary and add more if when it's called for. Adding things here wouldn't be breaking, right?

**Review by pacrob — APPROVED** [review]
LGTM! Nit/question - I've been grouping async functions together in the lower half of a file, separated with `# --- async --- #`. Do we want to continue that theme here? I think it's cleaner in an overall organization sense, but there's merit to having sync and async versions visually next to each other if they need changes in the future. :shrug:

**Comment by kclowes** [comment]
> I've been grouping async functions together in the lower half of a file, separated with # --- async --- #. Do we want to continue that theme here?

oops, yes! Thanks for the reminder!

**Review by fselmo — APPROVED** [review]
This lgtm! ~~I'd piggy back what @pacrob said about splitting out the async functions and say that might be useful to do inside the `utils.py` file as well for better readability~~ [oops that's the only file I see for that hah]. But that's a nit.

**Pull request merged**